From 1618c44776ec300a9be947dac5e6ed2e5120e3f2 Mon Sep 17 00:00:00 2001
From: Denis Kirjanov <denis.kirjanov@suse.com>
Date: Tue, 28 Feb 2023 18:17:51 +0300
Subject: [PATCH] sock.h: hide new member
Patch-mainline: Never (kABI fixup)
References: bsc#1194535 CVE-2021-4203

Put a new member into the hole in struct sock (found by pahole)

Signed-off-by: Denis Kirjanov <dkirjanov@suse.de>
---
 include/net/sock.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/net/sock.h b/include/net/sock.h
index 890e7b01cae0..037888ec8ccf 100644
--- a/include/net/sock.h
+++ b/include/net/sock.h
@@ -451,7 +451,9 @@ struct sock {
 	u32			sk_ack_backlog;
 	u32			sk_max_ack_backlog;
 	kuid_t			sk_uid;
+#ifndef __GENKSYMS__
 	spinlock_t		sk_peer_lock;
+#endif
 	struct pid		*sk_peer_pid;
 	const struct cred	*sk_peer_cred;
 
--
2.35.3

